### PR TITLE
GH-83 adding primary button component

### PR DESCRIPTION
**Pull request by AlexGriffith10**
This is adding the primary button component as described in GH-83. The button allows a user to provide the button text they want used, the function that will be used on click, as well as whether or not it is disabled and an optional success message if they want to use that functionality. The video shows the format the ticket asked for as well as the disabled state, a state with no success message, and one button utilizing custom inputs. The various inputs for each button are also listed below. I debated on whether or not to make the success message a separate variant but decided against it to avoid having clients needing to rerender swap out the component if they wanted to utilize that functionality

Note: success message reverts to regular message after 3 seconds and if a success message is not provided it does not show the success state but falls back to original state. Also so the video doesn't confuse anyone I clicked the no success button scenario twice its not a double load or anything

https://github.com/eosnetworkfoundation/components/assets/32306786/2bda64dd-212d-4019-a0fb-2c081f9dd274

### CODE REVIEW

**Comment by AlexGriffith10** [comment]
These are the inputs used in the video if it helps folks visualize usage

```
    const buttonInputs: PrimaryButtonInputs = {
        buttonText: 'Receive 1 EOS',
        clickAction: testFunction,
        isDisabled: false,
        successMessage: '1 EOS Sent!',
    };
    const disabledButtonInputs: PrimaryButtonInputs = {
        buttonText: 'Receive 1 EOS',
        clickAction: testFunction,
        isDisabled: true,
        successMessage: '1 EOS Sent!',
    };
    const noSuccessButtonInputs: PrimaryButtonInputs = {
        buttonText: 'No success message',
        clickAction: testFunction,
        isDisabled: false,
    };
    const customButtonInputs: PrimaryButtonInputs = {
        buttonText: 'Test text for science',
        clickAction: testFunction,
        isDisabled: false,
        successMessage: 'Successfully did the thing!',
    };
    
     <PrimaryButton primaryButtonInputs={buttonInputs} />

File: src/routes/+page.svelte [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+        console.log('Hello World!');
+    }
+
+    const buttonInputs: PrimaryButtonInputs = {
+        buttonText: 'Receive 1 EOS',
+        clickAction: testFunction,
+        isDisabled: false,
+        successMessage: '1 EOS Sent!',
+    };
 </script>
```
Comment (AlexGriffith10) [inline src/routes/+page.svelte:16]: left this in as an example but can remove if we think it isn't needed

File: src/routes/+page.svelte [hunk 2]
```diff
@@ -8,6 +22,9 @@
 
 <div>
     <h1 class="text-3xl font-bold underline">Hello world!</h1>
+    <div class="m-2">
+        <PrimaryButton primaryButtonInputs={buttonInputs} />
+    </div>
```
Comment (AlexGriffith10) [inline src/routes/+page.svelte:27]: same as above

File: src/lib/images/Spinner.svelte [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+>
+    <path
+        d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
+        fill="#E5E7EB"
+    />
+    <path
+        d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
+        fill="currentColor"
+    />
+</svg>
```
Comment (AlexGriffith10) [inline src/lib/images/Spinner.svelte:17]: svg as svelte file for easy use similar example below

File: src/components/primaryButton/PrimaryButton.svelte [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<script lang="ts">
+    import type { PrimaryButtonInputs } from '../../lib/types/primaryButton/PrimaryButton';
+    import Spinner from '../../lib/images/Spinner.svelte';
+    import CheckMark from '../../lib/images/Vector.svelte';
+    export let primaryButtonInputs: PrimaryButtonInputs;
+    let isLoading = false;
+    let isSuccess = false;
+
+    const { buttonText, clickAction, isDisabled, successMessage } = primaryButtonInputs;
+    const defaultStyle =
```
Comment (juanm95) [inline src/components/primaryButton/PrimaryButton.svelte:10]: Can we combine the common classes to make it easier to see what's changing between the two?

**Review by juanm95 — APPROVED** [review]
Just one comment to improve maintainability and readability.

File: src/components/primaryButton/PrimaryButton.svelte (hunk 4 above)
Comment (AlexGriffith10) [reply src/components/primaryButton/PrimaryButton.svelte:10]: We can but the downside is its ugly concatenating the strings I had initially tried to only have one class that would just swap out the color but it ended up looking gross so I just went with two separate classes. I'll see if there's a better way to make it look less clunky though

**Pull request merged**